src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -0,0 +1,213 @@
+    {
+        private const string ImmutableExceptionMessage = "This container has already been built and cannot be modified.";
+        private readonly Lazy<IDictionary<string, IServiceScope>> _serviceScopes = new Lazy<IDictionary<string, IServiceScope>>(() => new Dictionary<string, IServiceScope>());
+        private IServiceCollection _serviceCollection;
+        private bool _isImmutable;
+        private IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with an <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">An instance of <see cref="IServiceCollection"/>.</param>
+        public MicrosoftDependencyResolver(IServiceCollection services = null)
+        {
+            _serviceCollection = services ?? new ServiceCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftDependencyResolver" /> class with a configured service Provider.
+        /// </summary>
+        /// <param name="serviceProvider">A ready to use service provider.</param>
+        public MicrosoftDependencyResolver(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            _serviceProvider = serviceProvider;
+            _isImmutable = true;
+        }
+
+        /// <inheritdoc />
+        public virtual object GetService(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            var sp = GetServiceProvider(contract);
+            return sp.GetRequiredService(serviceType);
+        }
+
+        /// <inheritdoc />
+        public virtual IEnumerable<object> GetServices(Type serviceType, string contract = null)
+        {
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            var sp = GetServiceProvider(contract);
+            return sp.GetServices(serviceType);
+        }
+
+        /// <inheritdoc />
+        public virtual void Register(Func<object> factory, Type serviceType, string contract = null)
+        {
+            if (_isImmutable)
+            {
+                throw new InvalidOperationException(ImmutableExceptionMessage);
+            }
+
+            if (serviceType == null)
+            {
+                throw new ArgumentNullException(nameof(serviceType));
+            }
+
+            if (contract != null)
+            {
+                _serviceCollection.AddScoped(serviceType, _ => factory());
+            }
+            else
+            {
+                _serviceCollection.AddTransient(serviceType, _ => factory());
```

Comment:
You don't seem to be doing anything with the contract? This probably needs to change to some sort of input into the scope.

Reply:
In MS DI, you can "register as scoped", not "register per scope".